akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2StreamHandling.scala

```diff
@@ -338,6 +338,7 @@ private[http2] trait Http2StreamHandling { self: GraphStageLogic with LogHelper
         totalBufferedData += d.payload.size // padding can be seen as instantly discarded
 
         if (outstandingConnectionLevelWindow < 0) {
+          buffer.shutdown()
```

Comment:
These seem a bit like ad-hoc solutions to a more general problem:

 * it should not be possible to leak SubSinkInlets / SubSourceOutlets if the stage that handles them is shutdown (akka-stream problem)
 * we somehow need to keep track of those substreams and we currently do it explicitly in the state. Should we other logic to close those down automatically if we move to CLOSED state?